go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/ipfs/go-log/v2
 
-go 1.23.8
+go 1.23
```

Comment:
In order for `go mod tidy` to not add a `toolchain`, a specific minimum version needs to be specified, such as `go 1.23.0` When this file is merged into this repo, running `go mod tidy` must not change its contents.

```suggestion
go 1.23.2
```

Reply:
The reason a specific version is required (unlike `go 1.23` required by go-datastore`) is because of what dependencies require. If you try to update the go.mod version to 1.23, will see this error:
```
go mod tidy -go 1.23
go: golang.org/x/sys@v0.33.0 requires go@1.23.0, but 1.23 is requested
```